File: jna/src/main/java/dark/matrix.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;
import com.sun.jna.ptr.PointerByReference;

import java.util.Arrays;
import java.util.List;

public class matrix extends Structure {
    int rows;
    int cols;
    PointerByReference vals;
    public matrix() {
        super();
    }

    public matrix(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("rows","cols","vals");
    }
}

File: jna/src/main/java/dark_kotlin/matrix.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure
import com.sun.jna.ptr.PointerByReference

class matrix : Structure {
    @JvmField val rows = 0
    @JvmField val cols = 0
    @JvmField val vals: PointerByReference? = null //float**

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("rows", "cols", "vals")
    }
}
